src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TEMP_RANGE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.TemperatureRange;
+
+
+/**
+ * Search a person or a list of person according to a name or a range of temperature.
+ */
+public class SearchPatientCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchpatient";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Search a patient in Covigent. "
```

Comment:
Perhaps you want to consider changing the message to "Search for patients who fulfill the given criteria in Covigent." since the output may be a list of patients.

Reply:
yup, can change to "search a patient or a list of patient ..."

Reply:
I think "list of patient/all patients that match the specified criteria" may be clearer?

Reply:
I will change it to "Search a patient or a list of patients with specific criteria"